aries_vcx/src/libindy/utils/signus.rs

```diff
@@ -17,3 +21,95 @@ pub async fn create_and_store_my_did(seed: Option<&str>, method_name: Option<&st
+        Ok(PoolMocks::get_next_pool_response())
+    } else {
+        did::replace_keys_start(get_wallet_handle(), did, "{}")
+            .map_err(VcxError::from)
+            .await
+    }
+}
+
+pub async fn libindy_replace_keys_apply(did: &str) -> VcxResult<()> {
+    if pool_mocks_enabled() {
+        warn!("libindy_replace_keys_apply >> retrieving pool mock response");
+        Ok(())
+    } else {
+        did::replace_keys_apply(get_wallet_handle(), did)
+            .map_err(VcxError::from)
+            .await
+    }
+}
+
+pub async fn get_verkey_from_wallet(did: &str) -> VcxResult<String> {
+    if PoolMocks::has_pool_mock_responses() {
+        warn!("get_verkey_from_wallet >> retrieving pool mock response");
+        Ok(PoolMocks::get_next_pool_response())
+    } else {
+        did::key_for_local_did(get_wallet_handle(), did)
+            .map_err(VcxError::from)
+            .await
+    }
+}
+
+pub async fn get_verkey_from_ledger(did: &str) -> VcxResult<String> {
+    let nym_response: String = ledger::get_nym(did).await?;
+    let nym_json: Value = serde_json::from_str(&nym_response)
+        .map_err(|err| VcxError::from_msg(VcxErrorKind::SerializationError, format!("Cannot deserialize {:?} into Value, err: {:?}", nym_response, err)))?;
+    let nym_data: String = nym_json["result"]["data"].as_str()
+        .ok_or(VcxError::from_msg(VcxErrorKind::SerializationError, format!("Cannot deserialize {:?} into String", nym_json["result"]["data"])))?.to_string();
+    let nym_data: Value = serde_json::from_str(&nym_data)
+        .map_err(|err| VcxError::from_msg(VcxErrorKind::SerializationError, format!("Cannot deserialize {:?} into Value, err: {:?}", nym_data, err)))?;
+    Ok(nym_data["verkey"].as_str()
+        .ok_or(VcxError::from_msg(VcxErrorKind::SerializationError, format!("Cannot deserialize {:?} into String", nym_data["verkey"])))?
+        .to_string())
+}
+
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    use crate::utils::devsetup::*;
+    use crate::utils::mockdata::mockdata_pool;
+
+    #[cfg(feature = "pool_tests")]
+    #[tokio::test]
+    async fn test_rotate_verkey() {
+        let _setup = SetupWithWalletAndAgency::init().await;
+        let (did, verkey) = ledger::add_new_did(None).await;
+        rotate_verkey(&did).await.unwrap();
+        let local_verkey = get_verkey_from_wallet(&did).await.unwrap();
+        let ledger_verkey = get_verkey_from_ledger(&did).await.unwrap();
+        assert_ne!(verkey, ledger_verkey);
+        assert_eq!(local_verkey, ledger_verkey);
+    }
+
+    #[cfg(feature = "general_test")]
+    #[tokio::test]
+    async fn test_rotate_verkey_fails() {
+        let _setup = SetupPoolMocks::init();
+        let did = settings::get_config_value(settings::CONFIG_INSTITUTION_DID).unwrap();
+        PoolMocks::set_next_pool_response(mockdata_pool::RESPONSE_REQNACK);
+        PoolMocks::set_next_pool_response(mockdata_pool::NYM_REQUEST_VALID);
```

Comment:
Why is this (and following TRUSTEE_VERKEY_VALID) needed? Isn't 1 error response sufficient for this test?